tests/datasource/fluent/data_asset/data_connector/test_filesystem_data_connector.py

```diff
@@ -832,73 +644,52 @@ def test_relative_base_directory_path(tmp_path_factory):
-#             "james_20200810_1003.csv",
-#             "alex_20200819_1300.csv",
-#         ],
-#     )
-#     my_data_connector_yaml = yaml.load(
-#         f"""
-#         class_name: FilesystemDataConnector
-#         datasource_name: test_environment
-#         base_directory: {base_directory}
-#         glob_directive: "*.csv"
-#         assets:
-#             my_filesystem_data_asset:
-#                 pattern: (.+)_(.+)_(.+)\\.csv
-#                 group_names:
-#                     - name
-#                     - timestamp
-#                     - price
-#         default_regex:
-#             pattern: (.+)_.+_.+\\.csv
-#             group_names:
-#                 - name
-#         sorters:
-#             - orderby: asc
-#               class_name: LexicographicSorter
-#               name: name
-#             - datetime_format: "%Y%m%d"
-#               orderby: desc
-#               class_name: DateTimeSorter
-#               name: timestamp
-#             - orderby: desc
-#               class_name: NumericSorter
-#               name: for_me_Me_Me
-#     """,
-#     )
-#     with pytest.raises(gx_exceptions.DataConnectorError):
-#         # noinspection PyUnusedLocal
-#         my_data_connector: FilesystemDataConnector = (
-#             instantiate_class_from_config(
-#                 config=my_data_connector_yaml,
-#                 runtime_environment={
-#                     "name": "fluent",
-#                     "execution_engine": PandasExecutionEngine(),
-#                 },
-#                 config_defaults={
-#                     "module_name": "great_expectations.datasource.data_connector"
-#                 },
-#             )
-#         )
-# TODO: <Alex>ALEX</Alex>
+@pytest.mark.filesystem
+def test_filesystem_data_connector_uses_batching_regex_from_batch_request(tmp_path_factory):
+    # arrange
+    base_directory = str(tmp_path_factory.mktemp("test_basic_instantiation"))
+    create_files_in_directory(
+        directory=base_directory,
+        file_name_list=[
+            "alpha-1.csv",
+            "alpha-2.csv",
+            "alpha-3.csv",
+            "dont-match.csv",
+            "this-either.csv",
+        ],
+    )
+
+    my_data_connector: DataConnector = FilesystemDataConnector(
+        datasource_name="my_file_path_datasource",
+        data_asset_name="my_filesystem_data_asset",
+        batching_regex=MATCH_ALL_PATTERN,  # this pattern will match all 5 files
+        base_directory=pathlib.Path(base_directory),
+        glob_directive="*.csv",
+    )
+
+    # act
+
+    # establish a baseline without a batching_regex
+    all_batch_definitions = my_data_connector.get_batch_definition_list(
+        BatchRequest(
+            datasource_name="my_file_path_datasource",
+            data_asset_name="my_filesystem_data_asset",
+            options={},
+            batching_regex=None,
+        )
+    )
+
+    # provide a batching regex which should match 3 of the 5 files
```

Comment:
I'd really like to see this test switched over to be a parameterized so that there's one pytest.param for no batching regex and one with the regex